flow/shared/postgres.go

```diff
@@ -313,18 +317,34 @@ func ParsePgArrayToStringSlice(data []byte, delim byte) []string {
 			}
 		case psUnquoted:
 			if ch == '\\' {
+				null = 0
 				ps = psUnquotedEscape
 			} else if ch == '"' {
+				null = 0
 				ps = psQuoted
 			} else if ch == delim || ch == '}' {
-				result = append(result, string(sb))
+				if null == 4 {
+					result = append(result, "")
+				} else {
+					result = append(result, string(sb))
+				}
 				sb = sb[:0]
+				null = 0
 				if ch == '}' {
 					ps = psSearch2
 				} else {
 					ps = psSearch
 				}
 			} else {
+				if null == 1 && ch == 'U' {
```

Comment:
2 potential ideas:
1. could wrap this in "if null != 0" since that should generally be the case & avoid comparisons. Not worth 2 lines to me
2. could avoid appending N,U,L,L into sb, having to backfill sb when it turns out we aren't parsing NULL. Seems complicated, & appending these characters isn't allocating since we're reusing buffer